### PR TITLE
Draft - Have /plugin and /plugins apis return issue tracker url and default branch

### PR DESCRIPTION
**Pull request by halkeye**
It updates however often the generate plugin data job is run~, but neither pieces of info update that often, so should be fine~.

Looks like it (https://ci.jenkins.io/job/Infra/job/plugin-site-api/job/generate-data/) runs every 10 min, so might need to find a way to get it some github credentials. Its not doing anything sensitive so could create a dedicated dumb user and use a PAT

Anyways, wanted to get some feedback to know if this is a good way to solve the problem.

### CODE REVIEW

File: src/main/java/io/jenkins/plugins/generate/parsers/GithubDataParser.java [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+      logger.error("Error writing scm data for " + plugin.getName() + ": " + scm.getLink());
+      return;
+    }
+
+    String githubOrganization = parsed.getPath().split("/")[1];
+    String githubRepo = parsed.getPath().split("/")[2];
+    GithubRepoInformation repoInfo = fetchGithubInfo.getInfoForRepo(githubOrganization, githubRepo);
+    if (repoInfo != null) {
+      if (StringUtils.trimToNull(scm.getIssues()) != null) {
+        if (repoInfo.hasGithubIssuesEnabled) {
```
Comment (daniel-beck) [inline src/main/java/io/jenkins/plugins/generate/parsers/GithubDataParser.java:75]: This is an incorrect assumption in two different ways:

* Plugins don't necessarily track issues in just one issue tracker (I would argue due to how we're historically set up, the default is two different trackers)
* If GitHub issues are enabled, that doesn't mean that the preferred tracker is GH issues either.

If you'd invert this check to see whether the Jira component `${pluginId}-plugin` exists, and claim a plugin tracks issues in Jira, people would come running to complain (and rightfully so). This is not much different.

Comment (daniel-beck) [reply src/main/java/io/jenkins/plugins/generate/parsers/GithubDataParser.java:75]: (At a minimum, the UI and model should to support a list of issue trackers, perhaps with some semantic meaning to the order, i.e. descending priority -- that will make it easily extensible to support a more sophisticated check for where issues are tracked.)

**Comment by daniel-beck** [comment]
> could create a dedicated dumb user and use a PAT

Not exactly something we like doing, but with some luck we'll be able to get away with a PAT without any scopes.

Alternatively, we could look into putting the file elsewhere, e.g. reports.jenkins.io, and have trusted.ci or infra.ci put it there.

**Comment by halkeye** [comment]
> If you'd invert this check to see whether the Jira component `${pluginId}-plugin` exists, and claim a plugin tracks issues in Jira, people would come running to complain (and rightfully so). This is not much different.

its absolutely a hard thing to pick. I know @timja's original implementation did if its enabled, use issues since by default for a while now, issues have been disabled. Checking components isn't hard to do, we do it already live in plugins api, but there's lots of plugins that don't have components setup, mostly older legacy ones.

> If GitHub issues are enabled, that doesn't mean that the preferred tracker is GH issues either.
>  Plugins don't necessarily track issues in just one issue tracker (I would argue due to how we're historically set up, the default is two different trackers)

Just because something was done that way, isn't a good reason to continue doing it. I know from a end user point of view, its super confusing right now where/how to report issues. This is pushing it in the right direction so end user can get a clear way of how to report issues.

The logic i went with (and i've since learned scm.getIssues() doesn't contain what I want)
* POM.xml override
* if https://github.com/jenkins-infra/update-center2/pull/461 decides to get merged, then update-center override
* If github issues enabled, use github
* Fallback to jira as final default. I'm actually thinking the api shouldn't include the full issues url/component name so its easier to apps to consume it, maybe github/npm style of `jira:configuration-as-code`, `github`, `github:custom-repo`, `other just random url`

> Alternatively, we could look into putting the file elsewhere, e.g. reports.jenkins.io, and have trusted.ci or infra.ci put it there.

Oh i didn't know about reports, that might be better. The codebase already supports github app tokens, so easy to setup in a trusted env.

**Comment by daniel-beck** [comment]
> Just because something was done that way, isn't a good reason to continue doing it. I know from a end user point of view, its super confusing right now where/how to report issues. This is pushing it in the right direction so end user can get a clear way of how to report issues.

I acknowledge this in

> perhaps with some semantic meaning to the order, i.e. descending priority

Which is a complicated way of saying "The UI should use the first link for reporting new issues".

This doesn't change the reality that issues exist in multiple places for a ton of components, and ignoring that will not be helpful: The simple act of enabling the GH issue tracker for an established plugin will basically wipe all issue metadata from the plugin site. Even if that's where new issues should be reported, for displaying existing issues that won't be helpful.

**Comment by zbynek** [comment]
This is no longer needed, plugin-site gets this data directly from Update Center.

**Pull request closed without merging**